qutebrowser/utils/urlutils.py

```diff
@@ -702,3 +703,12 @@ def proxy_from_url(url):
     if url.password():
         proxy.setPassword(url.password())
     return proxy
+
+
+def widened_hostnames(hostname: str) -> typing.Iterable[str]:
```

Comment:
It might be beneficial to cache this (even with a small size). If all settings are done per-domain, this is computed 4 times per request with the same url for no reason.